step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```diff
@@ -70,67 +103,168 @@ public TestSetStatusOverview getTestSetStatusOverview(Map<String, String> reposi
         return new TestSetStatusOverview();
     }
 
-    private AutomationPackage getAutomationPackageForContext(Map<String, String> repositoryParameters) {
-        // we expect, that there is only one automation package stored per context
-        AutomationPackageManager automationPackageManager = getContext(repositoryParameters);
-        return automationPackageManager.getAllAutomationPackages(null).findFirst().orElse(null);
-    }
+    private PackageExecutionContext getOrRestorePackageExecutionContext(Map<String, String> repositoryParameters, ObjectEnricher enricher, ObjectPredicate predicate) {
+        String apName = repositoryParameters.get(RepositoryObjectReference.AP_NAME);
+        String contextId = repositoryParameters.get(REPOSITORY_PARAM_CONTEXTID);
+
+        // Execution context can be created in-advance and shared between several plans
+        PackageExecutionContext current = sharedPackageExecutionContexts.get(contextId);
+        if (current == null) {
+            // Here we resolve the original AP file used for previous isolated execution and re-use it to create the execution context
+            Resource resource = getResource(contextId, apName);
+
+            File apFile = null;
+
+            ResourceRevisionFileHandle fileHandle = resourceManager.getResourceFile(resource.getId().toString());
+            if (fileHandle != null) {
+                apFile = fileHandle.getResourceFile();
+            }
+            updateLastExecution(resource);
```

Comment:
do we really want to update the last execution if the ap file was not found

Reply:
@david-stephan 
Now I've put this update after the check on `apFile==null`